src/coreclr/jit/gentree.cpp

```diff
@@ -21336,9 +21336,9 @@ GenTree* Compiler::gtNewSimdZeroNode(var_types   type,
     NamedIntrinsic intrinsic = NI_Illegal;
 
 #if defined(TARGET_XARCH)
-    intrinsic = (simdSize == 32) ? NI_Vector256_get_Zero : NI_Vector128_get_Zero;
+    intrinsic = (simdSize > 16) ? NI_Vector256_get_Zero : NI_Vector128_get_Zero;
 #elif defined(TARGET_ARM64)
-    intrinsic     = (simdSize == 16) ? NI_Vector128_get_Zero : NI_Vector64_get_Zero;
+    intrinsic     = (simdSize > 8) ? NI_Vector128_get_Zero : NI_Vector64_get_Zero;
```

Comment:
This change is necessary. If `simdSize` is `12`, it cannot be stored in a `Vector64`, it must be a `Vector128`. This fixes the test failure on Arm64 that I mentioned.

Reply:
We shouldn't need the same change for the x86 side. It was already correct as `simdSize == 32`.

We'll probably want to do a small audit to catch other places where `Vector3` can hit a path and accidentally pick a `Vector64` operation. I'll log a bug for that and likely do a quick pass.